### DISCUSSION TITLE
1 to M matching

### DISCUSSION THREAD

**Discussion by luca-merlo** [NSAPH-Software/CausalGPS#237] [Ideas]
Hi @Naeemkh,
In our project with @wxwx1993  we had to implement 1 to M matching starting from GPS matching. To do this in R, I used the `which.minn` function in the "doBy" package by modifying the original R code of @wxwx1993 .
I am wondering if this could be a useful feature to add to the "CausalGPS" package even though the `compute_closest_wgps` function is now implemented in c++.

Thanks,
Luca

**Comment by Naeemkh** [comment]
Hi @luca-merlo,

It sounds like a great contribution to the project. You have the flexibility to contribute either in C++ or R. If you look at the current setup, R functions call the C++ functions. Similarly, you could have your R function call another R function. 

For a smooth and efficient contribution process, please adhere to the following steps. If you encounter any issues or have queries, don't hesitate to get in touch:

1. Fork the package into your own GitHub account.
2. Initiate a new branch for your modifications.
3. Ensure your system setup is correctly configured to work with the package. You can verify this by running the tests using `devtools::test()`.
4. Proceed to implement your changes, then submit a Pull Request to the 'develop' branch.
5. If working on an R code, include an @author field to credit contributors to the feature (You and Xiao).
6. Make sure that the documentation accurately reflects the feature you added.
7. Include unit tests for the new feature to validate its functionality.
8. Please adhere to the provided styling guide for code consistency.

More details about the contribution process can be found [here](https://nsaph-software.github.io/CausalGPS/articles/Developers-Guide.html).

Best regards,
Naeem

**Comment by luca-merlo** [comment]
Thanks @Naeemkh. I will work on that and let you know when I am finished.

Best,
Luca

**Comment by Naeemkh** [comment]
Moving this to discussions.
